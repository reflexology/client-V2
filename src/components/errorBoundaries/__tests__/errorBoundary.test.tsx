import React from 'react';
import renderer from 'react-test-renderer';
import { mount } from 'enzyme';
import { MemoryRouter } from 'react-router-dom';

import ErrorBoundary, { ErrorBoundaryState } from '../errorBoundary';

describe('<ErrorBoundary />', () => {
  const tree = (children?: React.ReactNode) =>
    renderer
      .create(
        <MemoryRouter>
          <ErrorBoundary>{children}</ErrorBoundary>
        </MemoryRouter>
      )
      .toJSON();

  it('renders children if no error', () => {
    expect(
      tree(
        <div>
          <h1>I am inside the error boundary</h1>
        </div>
      )
    ).toMatchSnapshot();
  });

  it('renders error view if an error occurs', () => {
    const wrapper = mount<{}, ErrorBoundaryState>(
      <ErrorBoundary>
        <div>
          <h1>I am inside the error boundary</h1>
        </div>
      </ErrorBoundary>
    );

    wrapper.setState({
      hasError: true
    });

    expect(wrapper.text() === 'Something went wrong.').toBeTruthy();
  });
});
