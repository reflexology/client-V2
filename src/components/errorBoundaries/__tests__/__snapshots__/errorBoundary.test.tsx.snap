// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ErrorBoundary /> renders children if no error 1`] = `
<div>
  <h1>
    I am inside the error boundary
  </h1>
</div>
`;
